app/assets/javascripts/discourse/app/routes/preferences-profile.js

```diff
@@ -1,7 +1,24 @@
+import { action } from "@ember/object";
 import RestrictedUserRoute from "discourse/routes/restricted-user";
 
 export default class PreferencesProfile extends RestrictedUserRoute {
   setupController(controller, model) {
     controller.set("model", model);
   }
+
+  @action
+  willTransition(transition) {
+    super.willTransition(...arguments);
+
+    if (
+      this.controllerFor("preferences.profile").get(
```

Comment:
Hmm I would probably not rely on a controller property for this, see note about `_missingRequiredFields`, then you could do `model.missingRequiredFields.length > 0`.

I think also if admins are trying to get into an admin route we need to allow this...otherwise they can lock themselves and all other admins out of the admin routes.

Reply:
I realised in my other comment this actually just points to `user.needs_required_fields_check`, so let's rely on this here instead of digging into the controller